Mlem/Views/Shared/SelectTextView.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  SelectTextView.swift
+//  Mlem
+//
+//  Created by Sjmarf on 03/03/2024.
+//
+
+import SwiftUI
+import SwiftUIIntrospect
+import Dependencies
+
+struct SelectTextView: View {
+    @Dependency(\.hapticManager) var hapticManager
+    @Environment(\.dismiss) var dismiss
+    
+    let text: String
+    
+    var body: some View {
+        VStack(spacing: 10) {
+            HStack {
+                Spacer()
+                Button {
+                    let pasteboard = UIPasteboard.general
+                    pasteboard.string = text
+                    hapticManager.play(haptic: .lightSuccess, priority: .high)
+                    dismiss()
+                } label: {
+                    Label("Copy All", systemImage: Icons.copyFill)
+                        .font(.footnote)
+                        .fontWeight(.semibold)
+                }
+                .foregroundStyle(.white)
+                .frame(height: 30)
+                .padding(.horizontal, 12)
+                .background(Capsule().fill(Color.accentColor))
+                Button {
+                    dismiss()
+                } label: {
+                    Image(systemName: "xmark.circle.fill")
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(height: 30)
+                        .symbolRenderingMode(.palette)
+                        .foregroundStyle(.secondary, .secondary.opacity(0.2))
```

Comment:
This introduces an inconsistency with our image viewer sheet, which feels similar enough that we should keep them aligned. The close/cancel/dismiss action is also almost always in the top left--it's a bit weird to see it over on the right.

Reply:
I put it in the top right for consistency with what iOS uses (share sheet, maps, etc). As far as I can tell they use this design everywhere Apple isn't entirely consistent with "cancel" buttons, though - they appear to be in the top-right when possible, or in the top-left if a "done" action is placed in the top-right.

If we want to remain consistent, maybe the image close button would be better on the right too?

Reply:
Moving the image viewer close button works for me--I agree that the apple default seems to prefer the right unless it's taken by a send.

Reply:
Done 👍